src/main/java/org/jenkinsci/plugins/parallel_test_executor/SplitStep.java

```diff
@@ -62,32 +67,48 @@ public void setStage(String stage) {
         this.stage = stage;
     }
 
-    @Extension public static final class DescriptorImpl extends AbstractStepDescriptorImpl {
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        return new Execution(context, this);
+    }
 
-        public DescriptorImpl() {
-            super(Execution.class);
+    @Extension
+    public static final class DescriptorImpl extends StepDescriptor {
+
+        @Override
+        public Set<? extends Class<?>> getRequiredContext() {
+            return ImmutableSet.of(TaskListener.class, Run.class);
         }
 
-        @Override public String getFunctionName() {
+        @Override
+        public String getFunctionName() {
             return "splitTests";
         }
 
-        @Override public String getDisplayName() {
+        @Override
+        public String getDisplayName() {
             return "Split Test Runs";
         }
-
     }
 
-    public static final class Execution extends AbstractSynchronousStepExecution<List<?>> {
+    public static final class Execution extends SynchronousStepExecution {
```

Comment:
Also can now be made `private`, since it is no longer being instantiated via reflection.

Reply:
+1
I will make it `private`.
I had the impression that backward compatibility is always a high goal in the Jenkins ecosystem.
A user will not be able to use this class anymore when it's `private` (not judging the usefulness of this).

Reply:
This class was only `public` for technical reasons and not intended to be instantiated programmatically, so I think it is OK.